sw/otbn/crypto/rsa_verify_3072.s

```diff
@@ -335,6 +335,101 @@ montmul:
 
   ret
 
+/**
+ * Variable time 3072-bit modular exponentiation with exponent 3
+ *
+ * Returns: C = modexp(A,3) = mod M
+ *
+ * The squared Montgomery modulus RR and the Montgomery constant m0' have to
+ * be provided at the appropriate locations in dmem.
+ *
+ * Flags: Flags have no meaning beyond the scope of this subroutine.
+ *
+ * The base bignum A is expected in the input buffer, the result C is written
+ * to the output buffer. Note, that the content of the input buffer is
+ * modified during execution.
+ *
+ * @param[in]  dmem[m0inv] pointer to m0' in dmem
+ * @param[in]  dmem[rr] pointer to RR in dmem
+ * @param[in]  dmem[in_mod] pointer to first limb of modulus M in dmem
+ * @param[in]  dmem[in_buf] pointer to buffer with base bignum
+ * @param[in]  dmem[out_buf] pointer to output buffer
+ *
+ * clobbered registers: x2, x5 to x13, x16 to x21, x29
+                        w2, to w15, w24 to w31
+ * clobbered Flag Groups: FG0, FG1
+ */
+ .globl modexp_var_3072_3
+modexp_var_3072_3:
+  /* prepare all-zero reg */
+  bn.xor    w31, w31, w31
+
+  /* prepare pointers to temp regs */
+  li         x8, 4
+  li         x9, 3
+  li        x10, 4
+  li        x11, 2
+
+  /* set pointers to buffers */
+  la        x24, out_buf
+  la        x16, in_mod
+  la        x23, in_buf
+  la        x26, rr
+  la        x17, m0inv
+
+  /* convert input to Montgomery domain and store in dmem
+     dmem[out_buf] = montmul(dmem[in_buf], dmem[in_RR]) = A*R mod M */
+  addi      x19, x23, 0
+  addi      x20, x26, 0
+  addi      x21, x24, 0
+  jal       x1, montmul
+  loopi     12, 2
+    bn.sid    x8, 0(x21++)
+    addi      x8, x8, 1
+
+  /* Square the outbut buffer:
+     dmem[out_buf]  <= montmul(dmem[out_buf], dmem[out_buf]) = (A^2)*R mod M */
```

Comment:
extra space before <=?

Is this intended to be an assignment? You use = above. (You could do `:=` to distinguish from the congruence at the end of the line.)

Reply:
The `=` is a typo, comments here use `<=` to represent assignments.